### DISCUSSION TITLE
[Help needed] Conformer-Transducer Streaming ASR with Microphone

### DISCUSSION THREAD

**Discussion by ItsaiNk** [NVIDIA-NeMo/NeMo#5264] [Q&A]
Hi, I really need help. I trained a Conformer-Transducer model and I have to online streaming inference with the audio recorded by a microphone, like [Online_ASR_Microphone_Demo.ipynb](https://github.com/NVIDIA/NeMo/blob/stable/tutorials/asr/Online_ASR_Microphone_Demo.ipynb). I also read the tutorials [Streaming_ASR.ipynb](https://github.com/NVIDIA/NeMo/blob/stable/tutorials/asr/Streaming_ASR.ipynb) and [Buffered_Transducer_Inference.ipynb](https://colab.research.google.com/github/NVIDIA/NeMo/blob/stable/tutorials/asr/Buffered_Transducer_Inference.ipynb) but I'm not understanding how to combine all this example to performe what I need. 
Someone can help me please? Thanks!

**Comment by titu1994** [comment]
Microphone support is quite fragile, and those scripts cannot easily be translated to it. 

If you need it just for a demo, a very simple approach would be to use a gradio app in streaming mode. 

For example, you can extend the code here as needed - https://huggingface.co/spaces/smajumdar/nemo_conformer_rnnt_large_streaming

**Reply by ItsaiNk** [reply]
Thank you for the answer! There is a way to pass the audio stream and not save a wav file before trascription?

**Reply by titu1994** [reply]
Currently, there isn't support for that. But we're eventually designing a more seperable transcription API that can take tensors as inputs.
